internal/backends/error.go

```diff
@@ -65,6 +70,21 @@ func NewError(code ErrorCode, err error) *Error {
 	}
 }
 
+// NewErrorWithArgument creates a new backend error with argument to be passed to the handler.
+//
+// Code must not be 0. Err may be nil.
+func NewErrorWithArgument(code ErrorCode, err error, arg any) *Error {
```

Comment:
I'm not sure about this approach, please share your ideas if you know how to improve it.

Reply:
By removing it. We talked about backend implementation not returning error messages, yet we still do that there for some reason.